### PR TITLE
Fix function addCssClass typehint (string|null): ?string.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️

### CODE REVIEW

**Comment by terabytesoftw** [comment]
Nav.php (yii-bootstrap4).

```
/**
 * Check to see if a child item is active optionally activating the parent.
 *
 * @param array $items @see items
 * @param bool $active should the parent be active too
 *
 * @return array
 *
 * {@see items}
 */
protected function isChildActive(array $items, bool &$active): array
{
    foreach ($items as $i => $child) {
        if (is_array($child) && !ArrayHelper::getValue($child, 'visible', true)) {
            continue;
        }

        if ($this->isItemActive($child)) {
            ArrayHelper::setValue($items[$i], 'active', true);
            if ($this->activateParents) {
                $active = true;
            }
        }

        $childItems = ArrayHelper::getValue($child, 'items');

        if (is_array($childItems)) {
            $activeParent = false;
            $items[$i]['items'] = $this->isChildActive($childItems, $activeParent);

            if ($activeParent) {
                Html::addCssClass($items[$i]['options'], 'active');
                $active = true;
            }
        }
    }

    return $items;
}
```
```
public function testDeepActivateParents()
{
    ob_start();
    ob_implicit_flush(0);

    Nav::counter(0);

    echo Nav::widget()
        ->activateParents(true)
        ->items([
            [
                'label' => 'Dropdown',
                'items' => [
                    [
                        'label' => 'Sub-dropdown',
                        'items' => [
                            ['label' => 'Page', 'content' => 'Page', 'active' => true],
                        ],
                    ],
                ],
            ],
       ]);
}
```

**Comment by samdark** [comment]
Shouldn't we add a check instead?

```php
if (isset($items[$i]['options'])) {
    Html::addCssClass($items[$i]['options'], 'active');
}
```

**Comment by terabytesoftw** [comment]
Not because by not defining any option and passing null to the function he adds the active class.

**Comment by samdark** [comment]
You're right. Thanks.

File: src/Html.php [hunk 1]
```diff
@@ -1263,12 +1263,13 @@ public static function renderTagAttributes(array $attributes): string
...
+     * @param array|null $options the options to be modified.
      * @param string|array $class the CSS class(es) to be added
-     * @see mergeCssClasses()
-     * @see removeCssClass()
+     * 
+     * {@see mergeCssClasses()}
+     * {@see removeCssClass()}
      */
-    public static function addCssClass(array &$options, $class): void
+    public static function addCssClass(?array &$options, $class): void
```
Comment (rob006) [inline src/Html.php]: BTW: can we remove reference here? New options could be just returned by method, there is no need to pass options as reference here.

**Pull request merged**